lib/rules/padding-line-between-tags.js

```diff
@@ -0,0 +1,193 @@
+    loc: sibling.loc,
+    // @ts-ignore
+    fix(fixer) {
+      const start = endTag.range[1]
+      const end = sibling.range[0]
+      const paddingText = context.getSourceCode().text.slice(start, end)
+      const lastSpaces = splitLines(paddingText).pop()
+      return fixer.replaceTextRange([start, end], `\n${lastSpaces}`)
+    }
+  })
+}
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+/**
+ * @param {RuleContext} context
+ */
+function checkNewline(context) {
+  /** @type {Array<{blankLine: "always" | "never", prev: string, next: string}>} */
+  const configureList = context.options[0] || [
+    { blankLine: 'always', prev: '*', next: '*' }
+  ]
+
+  // Copied from padding-line-between-tags
+  if (!context.parserServices.getDocumentFragment) {
+    return
+  }
+  const df = context.parserServices.getDocumentFragment()
+  if (!df) {
+    return
+  }
+  const documentFragment = df
+
+  /**
+   * @returns {VElement[]}
+   */
+  function getTopLevelHTMLElements() {
+    return documentFragment.children.filter(utils.isVElement)
+  }
+
+  /**
+   * @param {VElement} block
+   */
+  return (block) => {
+    if (!block.parent.parent) {
+      return
+    }
+
+    const endTag = block.endTag || block.startTag
+    const lowerSiblings = block.parent.children
+      .filter(
+        (element) =>
+          element.type === 'VElement' && element.range !== block.range
+      )
+      .filter((sibling) => sibling.range[0] - endTag.range[1] >= 0)
+
+    if (lowerSiblings.length === 0) {
+      return
+    }
+
+    const closestSibling = /** @type {VElement} */ (lowerSiblings[0])
+
+    for (let i = configureList.length - 1; i >= 0; --i) {
+      const configure = configureList[i]
+      const matched =
+        (configure.prev === '*' || block.name === configure.prev) &&
+        (configure.next === '*' || closestSibling.name === configure.next)
+
+      if (matched) {
+        const lineDifference =
+          closestSibling.loc.start.line - endTag.loc.end.line
+        if (configure.blankLine === 'always') {
+          if (lineDifference === 1) {
+            insertNewLine(context, block, closestSibling)
+          } else if (lineDifference === 0) {
+            const templateTag = getTopLevelHTMLElements().find(
+              (element) => element.name === 'template'
+            )
+            context.report({
+              messageId: 'always',
+              loc: closestSibling.loc,
+              // @ts-ignore
+              fix(fixer) {
+                if (!templateTag) {
+                  return
+                }
+                const start = templateTag.range[0]
+                const end = endTag.range[0]
+                const paddingText = context
+                  .getSourceCode()
+                  .text.slice(start, end)
+                const lastSpaces = splitLines(paddingText).pop()
```

Comment:
I did some tests to understand this code, but the value it processed is not necessarily spaces.
For example, if the following code is auto-fixed, the string will be doubled.

```vue
<template>
  <div>
  Foo  <div /><div />
  </div>
</template>
```

↓

```vue
<template>
  <div>
  Foo  <div />

  Foo  <div />
  </div>
</template>
```

I think we can probably fix it by doing:

```js
const lastSpaces = /** @type {RegExpExecArray} */(/^\s*/.exec(context.getSourceCode().lines[endTag.loc.start.line-1]))[0]
```